File: my-folder/0199-binary-tree-right-side-view/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        #go with level order traversal
        #last element in every level will be the ride side view
        if not root:
            return []
        queue = deque([])
        queue.append(root)
        output = []
        while queue:
            length_queue = len(queue)
            for _ in range(length_queue):
                node = queue.popleft()
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            output.append(node.val)
        return output



        

File: my-folder/0199-binary-tree-right-side-view/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        if(root == null){
            return result;
        }

        Queue<TreeNode> que = new LinkedList<>();
        que.offer(root);
        while(!que.isEmpty()){
            int currSize = que.size();
            for(int i = 0; i < currSize; i++){
                
                TreeNode curr = que.poll();
                if(i == currSize - 1){
                    result.add(curr.val);
                }

                if(curr.left != null){
                    que.offer(curr.left);
                }

                if(curr.right != null){
                    que.offer(curr.right);
                }
            }
        }
        return result;
    }
}
